### PR TITLE
updates for 0.49 test3

### LINKED ISSUE #379: "format s, 42"--[oscparse] crash
https://github.com/pure-data/pure-data/blob/f83c88ba7ac68368248797ef0d0febc3526432bb/src/x_misc.c#L555

           if (*sp)
                typecode = *sp++;
            else if (argv[j].a_type == A_SYMBOL)
                typecode = 's';
            else typecode = 'f';
            if (typecode == 's')
                msgindex += ROUNDUPTO4(strlen(argv[j].a_w.w_symbol->s_name) + 1);

`oscformat` trusts the type code given in the format specifier `sp`, before checking the type of the atom.  When the type code is `'s'` and the atom is not a symbol, the `w_symbol` field is bogus and an invalid memory access crash results.

### PR DESCRIPTION
**Pull request by danomatika**
Some more running updates:

* catch malformed messages to [oscformat], fixes #379
* fixed spelling error in posted message
* fixed undo-related crashes with `donecanvasdialog`, fixes https://github.com/pure-data/pure-data/pull/467

### CODE REVIEW

File: src/x_misc.c [hunk 1]
```diff
@@ -559,7 +559,15 @@ static void oscformat_list(t_oscformat *x, t_symbol *s, int argc, t_atom *argv)
             typecode = 's';
         else typecode = 'f';
         if (typecode == 's')
-            msgindex += ROUNDUPTO4(strlen(argv[j].a_w.w_symbol->s_name) + 1);
+        {
+            if (argv[j].a_type == A_SYMBOL)
+                msgindex += ROUNDUPTO4(strlen(argv[j].a_w.w_symbol->s_name) + 1);
+            else
+            {
+                pd_error(x, "oscformat: expected symbol but got float");
```
Comment (umlaeute) [inline src/x_misc.c]: iirc this error message is consistent with similar errors (even though the "float" might also be a "gpointer").
probably a better wording would be `"oscformat: expected symbol as argument #%d" % (j,))`

Comment (danomatika) [reply src/x_misc.c]: I thought of that, but the types that oscformat takes are only floats and symbols anyway. The rest get coerced into a float.

Comment (danomatika) [reply src/x_misc.c]: I would use "oscformat: expected symbol for argument %d" but should the index print be 0 or 1-indexed? ... :)

Comment (danomatika) [reply src/x_misc.c]: Done. 1-indexed makes sense in this context.

**Comment by millerpuckette** [comment]
merged

**Pull request closed without merging**